File: Neato-SDK/neato-sdk-android/src/main/java/com/neatorobotics/sdk/android/robotservices/spotcleaning/SpotCleaningService.java
package com.neatorobotics.sdk.android.robotservices.spotcleaning;

import com.neatorobotics.sdk.android.nucleo.RobotConstants;
import com.neatorobotics.sdk.android.robotservices.cleaning.CleaningService;


/**
 * Neato
 * Created by Marco on 04/01/2017.
 * Copyright © 2016 Neato Robotics. All rights reserved.
 */

public abstract class SpotCleaningService extends CleaningService {

    public SpotCleaningService() {
        this.cleaningType = RobotConstants.ROBOT_CLEANING_CATEGORY_SPOT;
    }
}

File: Neato-SDK/neato-sdk-android/src/main/java/com/neatorobotics/sdk/android/robotservices/spotcleaning/SpotCleaningService.kt
/*
 * Copyright (c) 2019.
 * Neato Robotics Inc.
 */

package com.neatorobotics.sdk.android.robotservices.spotcleaning

import com.neatorobotics.sdk.android.models.CleaningCategory
import com.neatorobotics.sdk.android.models.Robot
import com.neatorobotics.sdk.android.models.hasService
import com.neatorobotics.sdk.android.robotservices.RobotServices
import com.neatorobotics.sdk.android.robotservices.cleaning.CleaningService


val Robot.spotCleaningService: SpotCleaningService?
    get() {
        if (this.state == null) return null
        return if (this.hasService(RobotServices.SERVICE_SPOT_CLEANING)) {
            SpotCleaningServiceFactory.get(this.state!!.availableServices[RobotServices.SERVICE_SPOT_CLEANING]!!)
        } else
            null
    }

abstract class SpotCleaningService : CleaningService() {
    init {
        this.cleaningType = CleaningCategory.SPOT
    }
}
